app/controllers/users_controller.rb

```diff
@@ -44,12 +57,23 @@ def add_contact
     redirect_to root_path
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
   def search
-    @users = User.search(params[:search]).where.not(id: current_user.id)
-    @users_to_add = @users.reject do |user|
-      current_user.sent_contact_request?(user)
+    use_wildcards = true
+    @users = Set[]
+    @contacts = Set[]
+    if current_user && params[:search]
+      @user = User.find(current_user.id)
+      @users = search_record(params[:search], User, use_wildcards)
+      @contacts = search_record(params[:search], @user.contacts, use_wildcards)
```

Comment:
please use `current_user.sent_contact_request?` here because we not only want to send no second request to a user that is already our contact but also not to someone with a pending request

Reply:
Removed this search logic.